crates/signer-gcp/src/signer.rs

```diff
@@ -0,0 +1,348 @@
+use alloy_primitives::{hex, Address, B256};
+use alloy_signer::{Result, Signature, Signer};
+use async_trait::async_trait;
+use gcloud_sdk::{
+    google::cloud::kms::{
+        self,
+        v1::{
+            key_management_service_client::KeyManagementServiceClient, AsymmetricSignRequest,
+            GetPublicKeyRequest, PublicKey,
+        },
+    },
+    GoogleApi, GoogleAuthMiddleware,
+};
+use k256::ecdsa::{self, RecoveryId, VerifyingKey};
+use spki::DecodePublicKey;
+use std::{fmt, fmt::Debug};
+use thiserror::Error;
+use tonic::Request;
+
+type Client = GoogleApi<KeyManagementServiceClient<GoogleAuthMiddleware>>;
+
+/// Reference to a GCP KeyRing.
+#[derive(Clone, Debug)]
+pub struct GcpKeyRingRef {
```

Comment:
check my understanding: 
- the `GcpKeyRingRef` specifies a keyring by project, location, and name
- it is extended with a `key_id` and a `version` to form a full key specifier
- that key specifier is passed in the requests

we may want to clean it up with something like this? 

```
struct KeySpecifier { keyring: GcpKeyRingRef, key_id: String, version: u64 }

impl KeySpecifier { 
    fn to_key_version_ref(&self) -> String {
        // current behavior of `GcpKeyRingRef::to_key_version_ref`
    }
}
```

Reply:
yeah, that matches my understanding- eventually all these parameters are just used to construct a single string so have implemented `KeySpecifier(String)` since the inputs are never used individually